src/include/storage/garbage_collector.h

```diff
@@ -51,19 +56,145 @@ class GarbageCollector {
    */
   uint32_t ProcessUnlinkQueue();
 
+  /**
+   * Given a UndoRecord that has been deemed safe to unlink by the GC, attempts to remove it from the version chain.
+   * It's possible that this process will fail because the GC is conservative with conflicts. If the UndoRecord in the
+   * version chain to be updated in order to unlink the target UndoRecord is not yet committed, we will fail and
+   * expect this txn to be requeued and we'll try again on the next GC invocation, hopefully after the conflicting txn
+   * is either committed or aborted.
+   * @param undo_record UndoRecord to be unlinked
+   * @param active_txns list of timestamps of running transactions
+   * @return true, if the undo record was unlinked
+   */
+  bool ProcessUndoRecord(UndoRecord *undo_record, std::vector<transaction::timestamp_t> *active_txns);
```

Comment:
It seems that active_txns is not to be modified (in the function definition, it becomes a pointer to const values anyways), maybe consider const std::vector<transaction::timestamp_t> &active_txns

Reply:
`clang-tidy` recommends not to put `const` in the header files. The `const` declaration is present in the .cpp file.